extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -328,10 +336,57 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		});
 		if (uris && uris.length > 0) {
 			let bookPath = uris[0];
-			await this.openBook(bookPath.fsPath);
+			await this.openBook(bookPath.fsPath, undefined, true);
 		}
 	}
 
+	public async openNotebookFolder(): Promise<void> {
+		const allFilesFilter = loc.allFiles;
+		let filter: any = {};
+		filter[allFilesFilter] = '*';
+		let uris = await vscode.window.showOpenDialog({
+			filters: filter,
+			canSelectFiles: false,
+			canSelectMany: false,
+			canSelectFolders: true,
+			openLabel: loc.labelSelectFolder
+		});
+		if (uris && uris.length > 0) {
+			await this.loadNotebooksInFolder(uris[0]?.fsPath);
+		}
+	}
+
+	private async loadNotebooksInFolder(folderPath: string) {
+		let bookCollection = await this.getNotebooksInTree(folderPath);
+		for (let i = 0; i < bookCollection.bookPaths.length; i++) {
+			await this.openBook(bookCollection.bookPaths[i], undefined, false);
+		}
+		for (let i = 0; i < bookCollection.notebookPaths.length; i++) {
+			await this.openBook(bookCollection.notebookPaths[i], undefined, false, true);
+		}
+	}
+
+	private async getNotebooksInTree(folderPath: string): Promise<{ notebookPaths: string[]; bookPaths: string[]; }> {
```

Comment:
please make this into an interface

Reply:
Which part? The method itself or the return type?

Reply:
return type, sorry 😄